docs/source/security/cors.mdx

```diff
@@ -3,9 +3,7 @@ title: Configuring CORS
 description: Control access to your server's resources
 ---
 
-> ⚠️ New in Apollo Server 3.7: we **highly recommend** that all users pass `csrfPrevention: true` to `new ApolloServer()` to protect your server from CSRF and XS-Search attacks. **This is especially important if you integrate `graphql-upload` into your server.** If you enable this feature, any clients that send operations via `GET` requests or multipart upload requests must include a non-empty `Apollo-Require-Preflight` (or any other non-special-cased) header in that request. For more information, see [Preventing Cross-Site Request Forgery (CSRF)](#preventing-cross-site-request-forgery-csrf).
-
-> ⚠️ `graphql-upload` has a known CSRF vulnerability and is turned on by default in Apollo Server 2, specifically versions `< 2.25.4`. If you are using Apollo Server 2 and do _not_ use uploads, you can prevent this by disabling uploads by passing `uploads: false` to `new ApolloServer()` or by upgrading to version `^2.25.4`. If you _do_ use uploads, you **must** update to Apollo Server 3.7 or later and enable CSRF prevention in order to use this feature safely.
+> 📣 By default, Apollo Server 4 ships with a feature protecting from CSRF and XS-Search attacks. This feature requires that any client sending operations via `GET` or multipart upload requests _must_ include a non-empty `Apollo-Require-Preflight` (or another non-special-cased) header in that request. For more information, see [Preventing Cross-Site Request Forgery (CSRF)](#preventing-cross-site-request-forgery-csrf).
 
 [Cross-Origin Resource Sharing](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) (CORS) is an HTTP-header-based protocol that enables a server to dictate which origins can access its resources. Put another way, your server can specify which websites can tell a user's browser to talk to your server, and precisely which types of HTTP requests are allowed.
 
```

Comment:
(Next line)

This is more subtle now. It's:

- startStandaloneServer's behavior is to enable any website on the internet to tell a user's browser to connect to your server. It's not actually configurable, so it's not "the default". (Or well, I guess maaaybe you can tweak cors headers manually in your context function or a plugin or something if you want.)
- Otherwise, Apollo Server doesn't actually have any CORS behavior, but our *docs* encourage you to set that behavior by default.

Reply:
Oh, such a great catch! Okay, I fiddled around with this a bit and came up with something that I feel good about:

> The [`startStandaloneServer` function's](../migration#migrate-from-apollo-server) CORS configuration is _unalterable_ and enables any website on the internet to tell a user's browser to connect to your server. [Depending on your use case](#choosing-cors-options-for-your-project), you might need to further [customize your CORS behavior](#passing-credentials-with-cors) to ensure your server's security. To do so, you'll first need to swap to using [`expressMiddleware`](../migration#migrate-from-apollo-server-express) (or any other Apollo Server integration).